src/screens/Upload.js

```diff
@@ -1,57 +1,109 @@
-import React, {useContext} from 'react';
-import {View, StyleSheet, Text, TouchableOpacity} from 'react-native';
-import auth from '@react-native-firebase/auth';
-import {AuthContext} from '../navigation/AuthNavigator';
-
-export default function UploadScreen() {
-  const user = useContext(AuthContext);
-
-  async function logOut() {
-    try {
-      await auth().signOut();
-    } catch (e) {
-      console.error(e);
-    }
+import React from 'react';
+import {
+  Image,
+  PixelRatio,
+  StyleSheet,
+  Text,
+  TouchableOpacity,
+  View,
+} from 'react-native';
+import firestore from '@react-native-firebase/firestore';
+import uuid from 'uuid'; 
+import ImagePicker from 'react-native-image-picker';
+
+export default class UploadScreen extends React.Component {
+  state = {
+    avatarSource: null,
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.selectPhotoTapped = this.selectPhotoTapped.bind(this);
+  }
+
+  uploadPost = post => {
+    const id = uuid.v4();
+    const uploadData = {
+      id: id,
+      postPhoto: post.photo,
+      postTitle: post.title,
+    };
+    return firestore()
+      .collection('posts')
+      .doc(id)
+      .set(uploadData);
   }
 
-  return (
-    <View style={styles.container}>
-      <Text style={styles.title}>Upload Screen</Text>
-      <Text style={styles.title}>Welcome {user.uid}!</Text>
-      <TouchableOpacity style={styles.button} onPress={logOut}>
-        <Text style={styles.buttonText}>Sign out</Text>
-      </TouchableOpacity>
-    </View>
-  );
+  selectPhotoTapped() {
+    const options = {
+      quality: 1.0,
+      maxWidth: 500,
+      maxHeight: 500,
+      storageOptions: {
+        skipBackup: true,
+      },
+    };
+
+    ImagePicker.showImagePicker(options, response => {
+      console.log('Response = ', response);
+
+      if (response.didCancel) {
+        console.log('User cancelled photo picker');
+      } else if (response.error) {
+        console.log('ImagePicker Error: ', response.error);
+      } else if (response.customButton) {
+        console.log('User tapped custom button: ', response.customButton);
+      } else {
+        let source = { uri: response.uri };
+
+        // You can also display the image using data:
+        // let source = { uri: 'data:image/jpeg;base64,' + response.data };
+
+        this.setState({
+          avatarSource: source,
+        });
+      }
+    });
+  }
+
+
+  render() {
+    return (
+      <View style={styles.container}>
+        <TouchableOpacity onPress={this.selectPhotoTapped.bind(this)}>
```

Comment:
Maybe this should be in another file/component so if we want to edit it in the future we can?